src/mail/model/MailUtils.ts

```diff
@@ -303,6 +293,60 @@ export function markMails(entityClient: EntityClient, mails: Mail[], unread: boo
 	).then(noOp)
 }
 
+/**
+ * Check if all mails in the selection are drafts. If there are mixed drafts and non-drafts or the array is empty, return null. Otherwise, return true if drafts and false if not.
```

Comment:
The documentation on this is unclear, and the name and return value kind of obscure what this function is actually checking. This should return an enum, something like `Mixed`, `Drafts`, `NonDrafts`. I think the empty list case could probably be considered `Mixed`, or just make it an error and require you check before hand? 

Disregarding all of that, it does too much, and we only ever check the return value for `null`, so if all we care about is whether or not it's mixed/empty, then that's all we need to check
```
function containsDraftsAndNonDrafts(mails: ReadonlyArray<Mail>): boolean {
    mails.length === 0
    || (
        mails.some(mail => mail.state === MailState.DRAFT) 
        && mails.some(mail => mail.state !== MailState.DRAFT)
    )
}
````

if we find ourselved caring about the specific cases of all drafts or all mails, then we can just check those individually as needed.